#159987727 Add Favourite Article functionality

d-beloved opened the pull request:
#### What does this PR do?
- Add ability for a user to be able to favorite certain articles
- List the articles favorited by a user
- deletes articles favorited by a user

#### How should this be manually tested?
Run 'npm install'
create a .env using the .env.sample template
Run 'npm run migrate'
Run 'npm run start:dev'
use postman to test:
```routes added```
post: /api/articles/:articlesId/favorite
delete: /api/articles/:articlesId/favorite
get: /api/user/articles/favorite

#### What are the relevant pivotal tracker stories?
 #159987727

d-beloved committed:
feat(event): add favorite articles function

- authenticated user have favorite articles
- can check the list of favorite articles they have

[Delivers #159987727]

d-beloved committed:
fix: fix merge conflict

d-beloved committed:
fix: fix hound errors

gonuoha reviewed (changes requested):
please change you error responses to conform to the format we have been using ... {
status: 'error',
errors: {message: '...'}
}

d-beloved committed:
fix: bugFix

- make changes according to feedback on PR
- add more test for increasing coverage

[Delivers #159987727]

emekafredy reviewed (changes requested):
Please make sure to update the swagger docs file with your endpoints. reach out to @lauragift21 on this if you need help.

aknwosu commented on server/routes/api/favoriteArticle.js:
```diff
@@ -0,0 +1,19 @@
+import { Router } from 'express';
+import favoriteArticleController from '../../controllers/favoriteArticle';
+import auth from '../../middleware/auth';
```
import {authenticateUser} from '../../middleware/auth';

d-beloved replied:
Thanks, I've effected this change

aknwosu commented on server/routes/api/favoriteArticle.js:
```diff
@@ -0,0 +1,19 @@
+const router = Router();
+
+// methods that authenticates the user
+const { authenticateUser } = auth;
```
remove this

aknwosu replied:
you can remove this after modifying line 2

d-beloved committed:
fix: make changes according to PR comments

- increase coverage by writing more tests

emekafredy reviewed (approved):
LGTM
